src/types.rs

```diff
@@ -167,3 +171,35 @@ pub struct GetNetworkInfoResult {
     pub local_addresses: Vec<GetNetworkInfoResultAddress>,
     pub warnings: String,
 }
+
+#[derive(Serialize, Clone, Eq, PartialEq, Debug, Hash)]
+pub struct DescriptorChecksum(pub String);
+
+impl FromStr for DescriptorChecksum {
+    type Err = BwtError;
+    fn from_str(s: &str) -> Result<DescriptorChecksum, Self::Err> {
+        Ok(DescriptorChecksum(s.to_string()))
+    }
+}
+
+#[derive(Serialize, Clone, Eq, PartialEq, Debug, Hash)]
+pub struct Descriptor(pub String);
+
+impl Descriptor {
+    pub fn new(descriptor: &str, rpc: &RpcClient) -> Result<Self, BwtError> {
+        // TODO: what to do about non-ranged descriptors?
+        let info = rpc.get_descriptor_info(descriptor).unwrap();
+        Ok(Self(info.descriptor))
+    }
+
+    pub fn checksum(&self) -> DescriptorChecksum {
+        // This assumes that the descriptor is legit ...
+        DescriptorChecksum(self.0.split("#").collect::<Vec<_>>()[1].to_string())
```

Comment:
This would cause a panic if the descriptor is somehow invalid. How about adding a separate checksum field to the `Descriptor` struct and populating it in the constructor? We're already getting the checksum as a separate field from `getdescriptorinfo`.

Reply:
Fixed